utils/build_profiler.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+if [ "$#" -ne 1 ]; then
+    echo "build_profiler.sh requires one argument - product_string"
+    exit 1
+fi
+
+product_string="$1"
+
+# Create and change to .build_profiling dir 
+if [ ! -d ".build_profiling" ]; then
+    mkdir .build_profiling
+    if [ $? -ne 0 ]; then
+        >&2 echo "Creating the .build_profiling directory failed"
+        exit 1
+    fi
+fi
+
+cd .build_profiling
+if [ $? -ne 0 ]; then
+    >&2 echo "Changing to the .build_profiling directory failed"
+    exit 1
+fi
+
+# Create and change to product_string dir 
+if [ ! -d "$product_string" ]; then
+    mkdir "$product_string"
+    if [ $? -ne 0 ]; then
+        >&2 echo "Creating the $product_string directory failed"
+        exit 1
+    fi
+fi
+
+cd "$product_string"
+if [ $? -ne 0 ]; then
+    >&2 echo "Changing to the $product_string directory failed"
+    exit 1
+fi
+
+# Set log_number
+if [ -f "0.ninja_log" ]; then
+    # set log number to the number of the latest log file and add 1
+    log_number=$(ls *.ninja_log | cut -d'.' -f1 | sort -rn | head -n 1)
+    ((log_number=log_number+1))
+    echo "-----\nComparing logfile $log_number to baseline 0..."
```

Comment:
This will print the `\n` literally instead of a newline. You should use `echo -e` or maybe `printf`.